includes/truelayer-functions.php

```diff
@@ -16,7 +16,11 @@
  * @return void
  */
 function truelayer_print_error_message( $wp_error ) {
-	wc_print_notice( $wp_error->get_error_message(), 'error' );
+	if ( is_ajax() ) {
+		wc_add_notice( $wp_error->get_error_message(), 'error' );
+	} else {
+		wc_print_notice( $wp_error->get_error_message(), 'error' );
+	}
 }
```

Comment:
I don't agree that this is a better solution. Instead of a single if else, you have 4 if statements that needs to be checked, along with a foreach. We also wont get more than one error message from Truelayer since it will be thrown as an exception with a single message.

Also by making it a dynamic function call unnecessarily you are removing any form of type completion, deprecation checks and IntelliSense. A better solution if multiple error messages where possible would be to keep the same if check like we have here, and have a helper method to get the error message that can be called in both cases. Dynamic function calls can be good in some cases, but this in my opinion would be making the code more complex for less performance and readability. I'm not sure what the gain would be at all really.